version: '3.8'

name: rtbs
services:
  web:
    container_name: rtbs_web
    build: ./src
    command: python manage.py runserver 0.0.0.0:5001
    volumes:
      - ./src/application/:/usr/src/application/
    ports:
      - 5001:5001
    env_file:
      - ./.env
    depends_on:
      - database
    links:
      - database
  database:
    image: mysql:9.3.0
    container_name: rtbs_mysql
    restart: always
    volumes:
      - mysql_data:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${MYSQL_DATABASE}
      - MYSQL_USER=${MYSQL_USER}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD}
    ports:
      - 3306:3306
    healthcheck:
      test: ["CMD", "mysql", "-h", "localhost", "-u", "root", "-p ${MYSQL_ROOT_PASSWORD}", "-e", "SELECT 1"]
      timeout: 20s
      retries: 10

volumes:
  mysql_data:
